CONTRIBUTING.md

```diff
@@ -237,18 +237,18 @@ Please note that if adding a runtime guarded feature, your [release notes](chang
 * Typically we try to turn around reviews within one business day.
 * See [OWNERS.md](OWNERS.md) for the current list of maintainers.
 * It is generally expected that a senior maintainer should review every PR to
-  core code. Test-only or extension-only changes need only be reviewed by a
-  maintainer, or senior extension maintainer.
+  core code. Changes which only touch tests, extensions, tools, docs or comments
```

Comment:
`extensions` is a bit vague here, since we split many core feature as extensions, e.g. HCM, TLS. Let's say contrib extensions?

Reply:
+1, does it mean all extensions or non core extension?

Reply:
Can we clarify what is "core code" / "core extensions" refers to especially given the points made above

Reply:
I agree that some extensions are more critical than others. 

I also agree that indicating that via directory structure makes a lot of sense, so it's easy to tell the difference without needing to consult a list or something.

But my understanding is that some of the previously-core code was moved to extensions so we could use the existing build system mechanism to exclude it easily (intended for E-M, but may be useful in other cases).

Reply:
I don't think anyone but E-M cares about binary size the way we do, and if anyone does I'd assume they can depend on the same stripped targets that E-M does, and it'd all work out.

Reply:
Ok lets go with the text as is.  anything folks want me to move but the listener code?

Reply:
Probably all of the TCP/HTTP connection pool/upstream stuff? I haven't kept up to date with everything that was split out but I would say in general anything that is required for the basic HTTP proxy case, including HCM.

Reply:
HCM code is already in code but I can move the config files too, good call.

Reply:
SGTM